routes/chatbot.js

```diff
@@ -32,19 +41,35 @@ router.post('/', function (req, res, next) {
     glob.chatbot = body.glob.chatbot;
     utils.mixoutEnv(glob, envVars);
 
-    if ("deleteHook" == body.__action) {
+    if (!glob.chatbot.types) {
+        glob.chatbot.types = chatbotTypes;
+    }
+
+    if ("deleteTarget" === body.__action) {
         const index = Number(body.__object);
-        glob.chatbot.hookUrls.splice(index, 1);
-    } else if ("addHook" == body.__action) {
-        if (!glob.chatbot.hookUrls) {
-            glob.chatbot.hookUrls = [];
+        glob.chatbot.targets.splice(index, 1);
+    } else if ("addTarget" === body.__action) {
+        if (!glob.chatbot.targets) {
+            glob.chatbot.targets = [];
         }
-        glob.chatbot.hookUrls.push('https://url.to.your.slack/hookidentifierwhichisasecrect');
+        glob.chatbot.targets.push({
+            "type": chatbotTypes[0],
```

Comment:
Doesn't this always add a `slack` type target?

Reply:
Yes. Clicking on "add" will create a new target. The correct type and webhook is specified afterwards with an edit. This is also how it has worked before (but it just added a new generic hookUrl).